### PR TITLE
WT-4579 Track the newest durable timestamp for each page.

### PR DESCRIPTION
**Pull request by keithbostic**
I'm still debugging, but this is the basic shape of the change I expect to merge (there are still some interesting cases around fast-delete I need to solve).

### CODE REVIEW

**Comment by keithbostic** [comment]
autobuild: retest this please

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -4242,15 +4250,18 @@ __rec_col_var(WT_SESSION_IMPL *session,
 
 	/* Set the "last" values to cause failure if they're not set. */
 	last.value = r->last;
-	last.start_ts = last.stop_ts = WT_TS_NONE;
+	last.start_ts = WT_TS_MAX;
+	last.durable_ts = last.stop_ts = WT_TS_NONE;
```
Comment (agorrod) [inline src/reconcile/rec_write.c]: This change surprises me a bit. It's a change to the `start_ts`, and in other places `durable_ts` matches `start_ts`, not `stop_ts`. Similarly below.

**Comment by agorrod** [comment]
@bvpvamsikrishna I've added you as a reviewer here. Looking through this change made me think that we need a pretty strong way to ensure that commit and durable are always set if we don't go down the path of WT-4550 (handling durable only when we need to, rather than using it as the default most places).

File: src/reconcile/rec_write.c (hunk 1 above)
Comment (keithbostic) [reply src/reconcile/rec_write.c]: @agorrod, owed you a response:
> This change surprises me a bit. It's a change to the `start_ts`, and in other places `durable_ts` matches `start_ts`, not `stop_ts`. Similarly below.

This is unrelated to the rest of the changes. It's not a bug fix, either, it's just adding some measure of safety. The reason is so we (hopefully) fail if there's a bug and `last.start_ts` isn't set appropriately before being used, that is, a start timestamp of `WT_TS_NONE` is OK, one of `WT_TS_MAX` will fail.

**Comment by keithbostic** [comment]
@agorrod, I think this one is ready for a real review.

**Comment by michaelcahill** [comment]
Thanks @keithbostic, this lgtm.  I'll merge.

**Pull request merged**